LibTCPServer.emx

```diff
@@ -1212,35 +1221,96 @@ public:
 		str.append(result);
 		reply(str);
 	}
-
-private:
-	TCPServer_Actor& capsuleInst;
 };
 
-class LocalServerConnectionFactory: public TCPServerConnectionFactory
-	/// A factory for LocalServerConnection.
-{
+// Intercept incoming TCP requests and give them to the TCPServer for processing
+class ServerJob : public RTJob {
+	TCPServer_Actor* server;
+	const unsigned int maxSupportedClient;
+	unsigned int port;
+	long maxWaitForReply;
+	bool logToStdOut;
+	std::list<LocalServerConnection*> connections;
+
 public:
-	LocalServerConnectionFactory(TCPServer_Actor& context)
-		: capsuleInst(context)
-	{
+	ServerJob(TCPServer_Actor* s, unsigned int p, bool log, long maxWait, const unsigned int maxClient)
+	:server(s), port(p), logToStdOut(log), maxWaitForReply(maxWait), maxSupportedClient(maxClient) {}
+
+	virtual const char* name( void ) const override {
+		return "TCP Server Job";
 	}
-	
-	TCPServerConnection* createConnection(const StreamSocket& socket)
-	{
-		return new LocalServerConnection(socket, capsuleInst);
+
+	virtual void mainLoop( void ) override {
+		enter( RTJob::running );
+
+		RTIOMonitor ioMonitor;
+		RTTcpSocket listenSock;
+		listenSock.registerWith(&ioMonitor);
+		listenSock.setMaxPendingConnections(maxSupportedClient);
+		if (!listenSock.create() || !listenSock.listen(port))
+		{
+			std::cerr << "Failed to create TCP accept socket" << std::endl;
+			listenSock.close();
+			return;
+		}
+
+		do {
+			static const RTTimespec awhile(5L /* s */, 0L/*maxWaitForReply*1000*1000*/ /* ns */);
```

Comment:
I did some experiments with making this wait configurable so we need to clean-up this code.

Reply:
fixed by using config parameter maxWaitForReply.